Modification by SG

SylvainGa opened the pull request:
Addition of French localization
Addition of Pulse Ox as a data field
Modification to the AlwaysOn to reduce batterie drain

warmsound commented:
@SylvainGa thanks for all your hard work on this 🤝 . I'm starting the review, and will submit comments shortly.

warmsound commented on always-on-source/AlwaysOnDisplay.mc:
```diff
@@ -1,201 +1,212 @@
+		var hours = formattedTime[:hour];
+		var minutes = formattedTime[:min];
+		var amPmText = formattedTime[:amPm];
+		var colon = ":"; // SG Addition
```
I'm afraid I shouldn't allow the colon and font change to the always-on display: the bold hours/light minutes display is a defining characteristic of Crystal's display, so I'd prefer to keep this as is. The time display should be consistent between regular, and always-on.

For reference, note that this change also affects the newly-supported `venusq` and `venusqm` watches, which are currently broken:

![image](https://user-images.githubusercontent.com/6435959/103681895-faacc680-4f7f-11eb-8928-ea5d05711207.png)

The line endings have changed throughout this file.

Would you mind reversing changes to this file, and also to `resources-round-390x390/fonts/fonts.xml`?

We could consider making this change settings-based in future, although I'm not sure how to resolve the inconsistency with the regular time display, as there's no way a colon would fit for some devices.

warmsound replied:
Sorry, I've just noticed that the reason for making the always-on changes was to reduce battery drain. Is the saving due to using fewer fonts?

SylvainGa replied:
I've checked all the 'AlwaysOnHoursFont' and made sure the ':' is present.

warmsound commented on resources-fre/strings/strings.xml:
```diff
@@ -1,81 +1,81 @@
+	<string id="ThemeBlueDark">Bleu (sur fond noir)</string>
+	<string id="ThemePinkDark">Rose (sur fond noir)</string>
+	<string id="ThemeRedDark">Rouge (sur fond noir)</string>
+	<string id="ThemeGreenDark">Vert(sur fond noir)</string>
```
Missing space:

```suggestion
	<string id="ThemeGreenDark">Vert (sur fond noir)</string>
```

warmsound commented on resources-fre/strings/strings.xml:
```diff
@@ -89,16 +89,16 @@
 	<string id="Jun">Juin</string>
 	<string id="Jul">Juil</string>
-	<string id="Aug">Aout</string>
+	<string id="Aug">Août</string>
```
Many thanks for all these translations.

The accented `û` is not included in one or more date fonts, so appears incorrectly. Could you fix this?

![image](https://user-images.githubusercontent.com/6435959/103682380-ba9a1380-4f80-11eb-913d-9b357cc2884e.png)

SylvainGa replied:
Do you know which font is used or do simply cut and paste a standard 'u' and add a caret over it in PaintShopPro to make a û?

SylvainGa replied:
I think I've catched them all. Also fixed the ':' for low power mode Venu Sq.

warmsound commented on resources/strings/strings.bak:
```diff
@@ -0,0 +1,104 @@
+<strings>
```
Please remove this file, as I think it's not longer needed.

SylvainGa replied:
Ok, no problem

warmsound commented on source/DataFields.bak:
```diff
@@ -0,0 +1,730 @@
+using Toybox.WatchUi as Ui;
```
Please remove this file, as I think it's not longer needed.

warmsound commented on resources/settings/settings.xml:
```diff
@@ -88,6 +88,7 @@
 			<listEntry value="10">@Strings.SunriseSunset</listEntry>
 			<listEntry value="11">@Strings.Weather</listEntry>
 			<listEntry value="13">@Strings.Humidity</listEntry>
+			<listEntry value="14">@Strings.PulseOx</listEntry>
```
Crystal should only show settings that are actually supported by a given device (see how e.g. Temperature and Floors are handled). `currentOxygenSaturation` is [only supported](https://developer.garmin.com/connect-iq/api-docs/Toybox/Activity/Info.html#currentOxygenSaturation-var) by quite a limited set of devices - is there an easy way we can hide the setting from other devices?

SylvainGa replied:
I'll check

SylvainGa replied:
Do you mean like for example the resource file resources-approachs62 has "listEntry value="1">@Strings.FloorsClimbed</listEntry" in comments? If so, would that mean a specific resource file for each device that supports Pulse Ox?

SylvainGa replied:
Because it's supported by quite a few devices now:

Captain Marvel
D2™ Air
Darth Vader™
Descent™ Mk2 / Descent™ Mk2i
Descent™ Mk2 S
Enduro™
First Avenger
Forerunner® 245
Forerunner® 245 Music
Forerunner® 745
Forerunner® 945
Forerunner® 945 LTE
fēnix® 5X Plus
fēnix® 6 / 6 Solar / 6 Dual Power
fēnix® 6 Pro / 6 Sapphire / 6 Pro Solar / 6 Pro Dual Power / quatix® 6
fēnix® 6S / 6S Solar / 6S Dual Power
fēnix® 6S Pro / 6S Sapphire / 6S Pro Solar / 6S Pro Dual Power
fēnix® 6X Pro / 6X Sapphire / 6X Pro Solar / tactix® Delta Sapphire / Delta Solar / Delta Solar - Ballistics Edition / quatix® 6X / 6X Solar / 6X Dual Power
MARQ™ Adventurer
MARQ™ Athlete
MARQ™ Aviator
MARQ® Captain / MARQ® Captain: American Magic Edition
MARQ™ Commander
MARQ™ Driver
MARQ™ Expedition
MARQ™ Golfer
Rey™
Venu™
Venu™ 2
Venu™ 2S

warmsound commented on resources/strings/strings.xml:
```diff
@@ -76,6 +76,7 @@
 	<string id="Weather">Weather</string>
 	<string id="Humidity">Humidity</string>
 	<string id="Pressure">Pressure</string>
+	<string id="PulseOx">Pulse Ox</string>
```
Please also add this string to the remaining 19 translation files, so that it can be easily translated in future.

SylvainGa replied:
Done

warmsound commented on source/DataFields.mc:
```diff
@@ -368,6 +370,14 @@ class DataFields extends Ui.Drawable {
+				activityInfo = Activity.getActivityInfo();
+				sample = activityInfo != null and activityInfo has :currentOxygenSaturation ? activityInfo.currentOxygenSaturation : null;
+				if (sample != null) {
+					value = sample.format(INTEGER_FORMAT);
```
Do you think the PulseOx value should be shown with a trailing "%", like Humidity and Battery?

SylvainGa replied:
Would make sense, yes since it's a percentage

SylvainGa replied:
Done

warmsound reviewed (changes requested):
Do you have a vector version of the O2 icon? If so, I'd like to update the `crystal-icons.afdesign` Affinity Designer file after merge. If not, don't worry.

Thanks again for your work on this.

SylvainGa commented:
> 
> 
> Do you have a vector version of the O2 icon? If so, I'd like to update the `crystal-icons.afdesign` Affinity Designer file after merge. If not, don't worry.
> 
> Thanks again for your work on this. If you'd like any help addressing the comments, please let me know.

No, I'm sorry. I don't have a vectored graphic for the O2. I simply used PaintShop Pro and used the font Segoe UI to write O and 2 using two different font size.

warmsound commented:
@SylvainGa, would you like any help with this? I'd be happy to merge as is, and then I could make further changes before release, along the above lines. Your changes are valuable, so it'd be great to get them in 👍

SylvainGa commented:
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  Sorry, had a ski accident. Can you refresh what you want me to change again?                                                                                                                                              Envoyé via Boîte de réception BlackBerry Hub+ pour Android                                                                                                                                                                           De : notifications@github.comEnvoyé : 24 janvier 2021 06 h 03À : crystal-face@noreply.github.comRépondre à : reply@reply.github.comCc : gagnon.sylvain@gmail.com; mention@noreply.github.comObjet : Re: [warmsound/crystal-face] Modification by SG (#184)  

@SylvainGa, would you like any help with this? I'd be happy to merge as is, and then I could make further changes before release, along the above lines. Your changes are valuable, so it'd be great to get them in 👍



—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

warmsound commented:
Sorry to hear about your accident. Wishing you a speedy recovery, and thank you again for your work on this - there's no urgent hurry.

lrosenman commented:
I would love to see this make it into a release at some point.

SylvainGa commented:
Hi Larry, hope you're doing well. I haven't seen any movement for almost a
year in your crystal-face Garmin watchface. Are you still planning on
releasing a new version with the updated code?

Regards,
Sylvain.

Le mer. 9 juin 2021, à 22 h 46, Larry Rosenman ***@***.***> a
écrit :

> I would love to see this make it into a release at some point.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/warmsound/crystal-face/pull/184#issuecomment-858237333>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALYXESRJVSDBYQAK5ZTW5LLTSARR7ANCNFSM4VQ2A26Q>
> .
>

lrosenman commented:
I'm not the code owner.

@warmsound  is.

SylvainGa commented:
Hi, I hope you're doing well. I haven't seen any movement for almost a
year in your crystal-face Garmin watchface. Are you still planning on
releasing a new version with the updated code?

Regards,
Sylvain.

Le jeu. 28 janv. 2021, à 12 h 47, warmsound ***@***.***> a
écrit :

> Sorry to hear about your accident. Wishing you a speedy recovery, and
> thank you again for your work on this - there's no urgent hurry.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/warmsound/crystal-face/pull/184#issuecomment-769259794>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALYXEST5BGCL7EMAQHAXAGDS4GPKHANCNFSM4VQ2A26Q>
> .
>

SylvainGa commented:
Yeah, sorry. Replied to the wrong thread.

Le mar. 22 mars 2022, à 20 h 07, Larry Rosenman ***@***.***>
a écrit :

> I'm not the code owner.
>
> @warmsound <https://github.com/warmsound> is.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/warmsound/crystal-face/pull/184#issuecomment-1075769942>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALYXESW23TW25DED4YHUYYDVBJOETANCNFSM4VQ2A26Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
